crates/biome_cli/src/reporter/terminal.rs

```diff
@@ -90,13 +43,17 @@ impl<'a> ReporterVisitor for ConsoleReporterVisitor<'a> {
         }
 
         self.0.log(markup! {
-            {ConsoleTraversalSummary(execution.traversal_mode(), summary)}
+            {ConsoleTraversalSummary(execution.traversal_mode(), &summary)}
         });
 
         Ok(())
     }
 
-    fn report_diagnostics(&mut self, diagnostics_payload: &DiagnosticsPayload) -> io::Result<()> {
+    fn report_diagnostics(
+        &mut self,
+        execution: &Execution,
+        diagnostics_payload: DiagnosticsPayload,
```

Comment:
We must take ownership of the diagnostics because we need to convert them to serde diagnostics, and the `Error` type can't be cloned.